Autoscale on AzureManagedMachinePool panics

/kind bug

Enabling auto-scaling for Azure results in a panic.

```
I0120 13:43:57.942034       1 azuremanagedcontrolplane_controller.go:226] controllers.AzureManagedControlPlaneReconciler.reconcileNormal "msg"="Reconciling AzureManagedControlPlane" "AzureManagedControlPlane"={"name":"ci-asaks3433334","namespace":"sfd-ci-asaks3433334"} "controller"="azuremanagedcontrolplane" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedControlPlane" "name"="ci-asaks3433334" "namespace"="sfd-ci-asaks3433334" "reconcileID"="9a61c4d2-a36c-4313-8c70-3a59ad0e6f37" "x-ms-correlation-request-id"="8a5e8c07-b179-4c0f-ba3d-434c44bb9832"
I0120 13:43:57.942049       1 azuremanagedmachinepool_controller.go:246] controllers.AzureManagedMachinePoolReconciler.reconcileNormal "msg"="Reconciling AzureManagedMachinePool" "AzureManagedMachinePool"={"name":"ci-asaks3432502-0","namespace":"sfd-ci-asaks3432502"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="ci-asaks3432502-0" "namespace"="sfd-ci-asaks3432502" "reconcileID"="56871be0-af5f-4d93-af49-dfcaccc94100" "x-ms-correlation-request-id"="aa91f95e-8116-443f-ae69-34476b8911d7"
I0120 13:43:57.942299       1 azuremanagedmachinepool_controller.go:246] controllers.AzureManagedMachinePoolReconciler.reconcileNormal "msg"="Reconciling AzureManagedMachinePool" "AzureManagedMachinePool"={"name":"ci-asaks3433334-0","namespace":"sfd-ci-asaks3433334"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="ci-asaks3433334-0" "namespace"="sfd-ci-asaks3433334" "reconcileID"="1487ac04-8647-443c-a80d-bd3e52da0770" "x-ms-correlation-request-id"="2cc648ac-392d-4f9b-a9c3-5e70693a6632"
I0120 13:43:57.942644       1 azuremanagedmachinepool_reconciler.go:98] controllers.azureManagedMachinePoolService.Reconcile "msg"="reconciling managed machine pool" "AzureManagedMachinePool"={"name":"ci-asaks3432502-0","namespace":"sfd-ci-asaks3432502"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="ci-asaks3432502-0" "namespace"="sfd-ci-asaks3432502" "reconcileID"="56871be0-af5f-4d93-af49-dfcaccc94100" "x-ms-correlation-request-id"="aa91f95e-8116-443f-ae69-34476b8911d7"
I0120 13:43:57.943367       1 azuremanagedmachinepool_reconciler.go:98] controllers.azureManagedMachinePoolService.Reconcile "msg"="reconciling managed machine pool" "AzureManagedMachinePool"={"name":"ci-asaks3433334-0","namespace":"sfd-ci-asaks3433334"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="ci-asaks3433334-0" "namespace"="sfd-ci-asaks3433334" "reconcileID"="1487ac04-8647-443c-a80d-bd3e52da0770" "x-ms-correlation-request-id"="2cc648ac-392d-4f9b-a9c3-5e70693a6632"
I0120 13:43:59.644265       1 controller.go:117]  "msg"="Observed a panic in reconciler: assignment to entry in nil map" "AzureManagedMachinePool"={"name":"ci-asaks3432581-0","namespace":"sfd-ci-asaks3432581"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="ci-asaks3432581-0" "namespace"="sfd-ci-asaks3432581" "reconcileID"="b98fac59-fcd7-4f98-a240-07e0d6dbd57e"
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 1173 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x2000f20, 0x2686fd0})
	/usr/local/go/src/runtime/panic.go:884 +0x212
sigs.k8s.io/cluster-api-provider-azure/azure/scope.(*ManagedMachinePoolScope).SetCAPIMachinePoolAnnotation(0xc00138a0c0?, {0x237a301?, 0xc00138a5a0?}, {0x233c7a2, 0x4})
	/workspace/azure/scope/managedmachinepool.go:313 +0x49
sigs.k8s.io/cluster-api-provider-azure/azure/services/agentpools.(*Service).Reconcile(0xc0008ecd00, {0x26a6cb8?, 0xc00138a180?})
	/workspace/azure/services/agentpools/agentpools.go:88 +0x1bd
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*azureManagedMachinePoolService).Reconcile(0xc00138a0f0, {0x26a6cb8?, 0xc000ae5560?})
	/workspace/exp/controllers/azuremanagedmachinepool_reconciler.go:101 +0x12e
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*AzureManagedMachinePoolReconciler).reconcileNormal(0xc0009fa1c0, {0x26a6cb8?, 0xc0011ab8c0?}, 0xc000441db0)
	/workspace/exp/controllers/azuremanagedmachinepool_controller.go:260 +0x16f
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*AzureManagedMachinePoolReconciler).Reconcile(0xc0009fa1c0, {0x26a6cb8?, 0xc0011ab830?}, {{{0xc000bec750?, 0xa?}, {0xc000bec738?, 0x0?}}})
	/workspace/exp/controllers/azuremanagedmachinepool_controller.go:239 +0xaaf
sigs.k8s.io/cluster-api-provider-azure/pkg/coalescing.(*reconciler).Reconcile(0xc0009fa300, {0x26a6cb8?, 0xc0011ab770?}, {{{0xc000bec750?, 0x10?}, {0xc000bec738?, 0x40dc87?}}})
	/workspace/pkg/coalescing/reconciler.go:109 +0x3ee
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x26a6c10?, {0x26a6cb8?, 0xc0011ab770?}, {{{0xc000bec750?, 0x221f060?}, {0xc000bec738?, 0x404554?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000712dc0, {0x26a6c10, 0xc000bcfe00}, {0x20b57c0?, 0xc0010a5260?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000712dc0, {0x26a6c10, 0xc000bcfe00})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:230 +0x333
```

**What steps did you take and what happened:**
Enabling Azure Managed AKS cluster with `AzureMangedMachinePool` set to 
```
Spec:
  Enable Node Public IP:  false
  Max Pods:               100
  Mode:                   System
  Name:                   pool0
  Os Disk Size GB:        512
  Os Disk Type:           Managed
  Os Type:                Linux
  Scaling:
    Max Size:  5
    Min Size:  3
  Sku:         Standard_D8s_v3
```


**What did you expect to happen:**
The AzureManagedMachinePool to be provisioned, but it fails.


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- cluster-api-provider-azure version: v1.7.0
- Kubernetes version: (use `kubectl version`): 1.25.4 
- OS (e.g. from `/etc/os-release`): ubuntu:latest

--- a/azure/scope/managedmachinepool.go
+++ b/azure/scope/managedmachinepool.go
@@ -309,6 +309,9 @@ func (s *ManagedMachinePoolScope) SetCAPIMachinePoolReplicas(replicas *int32) {
 
 // SetCAPIMachinePoolAnnotation sets the specified annotation on the associated MachinePool.
 func (s *ManagedMachinePoolScope) SetCAPIMachinePoolAnnotation(key, value string) {
+	if s.MachinePool.Annotations == nil {
+		s.MachinePool.Annotations = make(map[string]string)
+	}
 	s.MachinePool.Annotations[key] = value
 }
 

